### PR TITLE
add support for user supplied custom launch_arguments

### PR DESCRIPTION
**Pull request by hjanuschka**
as requested in: https://github.com/fastlane/fastlane/issues/7907
support for user specified launch arguments, i think the feature is really usefull in many use-cases (and goes along with `snapshot`)

set launch arguments via Screengrabfile:
```ruby
launch_arguments: ["username hjanuschka", "build_type hero"]
```

get it in your test class

```java
{
    Bundle extras = InstrumentationRegistry.getArguments();
    String peerID = null;

    if ( extras != null ) {
        if ( extras.containsKey ( "username" ) ) {
            username = extras.getString("username");
            System.out.println("Username: " + username);
        } else {
            System.out.println("No username in extras");
        }
    } else {
        System.out.println("No extras");
    }
}
```

i got the sample from here: http://stackoverflow.com/a/34726783/1945875

tested the map/join with:

```ruby
launch_arguments = ["username asdfasf", "somethingelse asdf"]
puts launch_arguments.map { |item| '-e ' + item + ' ' }.join.to_s
exit
```

@asfalcone, @mfurtak  could you help me with review?

### CODE REVIEW

File: screengrab/lib/screengrab/runner.rb [hunk 1]
```diff
@@ -219,6 +219,7 @@ def run_tests(device_serial, test_classes_to_use, test_packages_to_use)
                               "-e endingLocale #{@config[:ending_locale].tr('-', '_')}"]
         instrument_command << "-e class #{test_classes_to_use.join(',')}" if test_classes_to_use
         instrument_command << "-e package #{test_packages_to_use.join(',')}" if test_packages_to_use
+        instrument_command << launch_arguments.map { |item| '-e ' + item + ' ' }.join.to_s if launch_arguments
```
Comment (lacostej) [inline screengrab/lib/screengrab/runner.rb]: Nitpick: wouldn't thi work as well?
```ruby
instrument_command << launch_arguments.map { |item| '-e ' + item }.join(' ') if launch_arguments
```

Comment (hjanuschka) [inline screengrab/lib/screengrab/runner.rb]: feedback addressed

**Review by lacostej — APPROVED** [review]
:+1:

File: screengrab/spec/runner_spec.rb [hunk 2]
```diff
@@ -24,6 +24,20 @@ def mock_adb_response(mock_response)
...
     let(:test_packages_to_use) { nil }
 
+    context "when launch arguments are specified" do
+      before do
+        config[:launch_arguments] = ["username hjanuschka", "build_type x500"]
+        config[:locales] = %w(en-US)
+        config[:ending_locale] = 'en-US'
+      end
+      it 'sets custom launch_arguments' do
+        # expect(FastlaneCore::CommandExecutor).to receive(:execute).with(/hjanuschka/)
```
Comment (mfurtak) [inline screengrab/spec/runner_spec.rb]: Please remove dead code 👍

**Comment by mfurtak** [comment]
Nice feature! It would be great if we could add information about it to the [docs](https://github.com/fastlane/docs/blob/master/docs/getting-started/android/screenshots.md) so that people can learn about it

**Comment by hjanuschka** [comment]
@mfurtak removed the dead code in #7917, and i ll do a docs PR

**Pull request merged**